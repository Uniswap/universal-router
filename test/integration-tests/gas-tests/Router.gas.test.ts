import { Router } from '../../../typechain'
import { expect } from '../shared/expect'
import { ALICE_ADDRESS } from '../shared/constants'
import { resetFork } from '../shared/mainnetForkHelpers'
import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
import hre from 'hardhat'
import deployRouter from '../shared/deployRouter'

const { ethers } = hre

describe('Router Gas Tests', () => {
  let alice: SignerWithAddress
  let router: Router

  beforeEach(async () => {
    await resetFork()
    alice = await ethers.getSigner(ALICE_ADDRESS)
    await hre.network.provider.request({
      method: 'hardhat_impersonateAccount',
      params: [ALICE_ADDRESS],
    })
    router = (await deployRouter()).connect(alice) as Router
  })

  it.only('gas: bytecode size', async () => {
    expect(((await router.provider.getCode(router.address)).length - 2) / 2).to.matchSnapshot()
  })
})
