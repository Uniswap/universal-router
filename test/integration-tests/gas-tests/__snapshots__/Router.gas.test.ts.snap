// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router Gas Tests gas: bytecode size 1`] = `14895`;

exports[`Router Gas Tests trading for NFTs gas: ERC20 --> ETH --> Seaport NFT 1`] = `
Object {
  "calldataByteLength": 2500,
  "gasUsed": 302999,
}
`;

exports[`Router Gas Tests trading for NFTs gas: ETH --> Seaport NFT 1`] = `
Object {
  "calldataByteLength": 2052,
  "gasUsed": 200368,
}
`;
